frontend/src/simulation/simulation.js

```diff
@@ -0,0 +1,247 @@
+                    value={this.state.budgetProposal[resource]}
+                    onChange={(e) => this.handleSliderOnChange(e, resource)}
+                />
+
+            </div>
+        ));
+
+        return(
+            <>
+                <div>
+                    {budgetOptions}
+                </div>
+
+                <div className="submit_button_container">
+                    <button
+
+                        type="submit"
+                        onClick={this.simulateCitizenResponse}
+                    >
+                        Submit Budget
+                    </button>
+                </div>
+
+                <div>
+                    {result} out of {this.props.population.length} people support your budget
+                </div>
+
+            </>
+        )
+    }
+}
+Budget.propTypes = {
+    population: PropTypes.array,
+};
+
+
+class AggregateData extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            categories: Object.keys(this.props.population[0]["traits"])
+        }
+    }
+
+    render() {
+        const aggregate_values = {};
+        for (let i = 0; i < this.state.categories.length; i++) {
+            let total = 0;
+            for (let j = 0; j < this.props.population.length; j++) {
+                total += this.props.population[j]["traits"][this.state.categories[i]]
+                //Above: index into the population to get a person, then that person's traits and
+                //then the value (true or false) of that trait
+            }
+            aggregate_values[[this.state.categories[i]]] = total/this.props.population.length;
+        }
+
+        return (
+            <table className="aggregate_data_table">
+                <thead>
+                    <tr>
+                        <th className="aggregate_data_table_header">
+                            Trait
+                        </th>
+                        <th className="aggregate_data_table_data">
+                            Percentage of Population
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {this.state.categories.map((category, key) => {
+                        return (
+                            <tr key={key} className="aggregate_data_table_rows">
+                                <td className="aggregate_data_table_data">
+                                    {category}
+                                </td>
+
+                                <td className="aggregate_data_table_data">
+                                    {(aggregate_values[category]*100).toFixed(1)}%
+                                </td>
+                            </tr>
+                        )
+                    })}
+                </tbody>
+            </table>
+        )
+    }
+}
+AggregateData.propTypes = {
+    population: PropTypes.array,
+};
+
+class MainView extends React.Component {
```

Comment:
Please rename this component into something descriptive of this whole mini-app. Remember it needs to live in a codebase with the other sims/games, too!